contracts/MetawoodAuction.sol

```diff
@@ -0,0 +1,330 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import "./interfaces/IMetawoodNFT.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "@openzeppelin/contracts/token/ERC1155/utils/ERC1155Holder.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+contract MetawoodAuction is ERC1155Holder, Ownable, ReentrancyGuard {
```

Comment:
Nice touch on the ERC1155 holder contract.
Please comment what are the other approaches here in other auction contracts ?
Do they follow the same or don't transfer the NFT from the seller until the seller accepts the bid

Reply:
Using the holder contract I read about in OpenZeppelin docs, they advised to use it. We discussed this flow in a call earlier, making it like traditional auctions.

Reply:
Okay cool

Reply:
OZ link for reference https://docs.openzeppelin.com/contracts/3.x/erc1155#sending-to-contracts